Tella/Scenes/Uwazi/ViewModel/AddTemplateVM.swift

```diff
@@ -68,9 +71,13 @@ class AddTemplateViewModel: ObservableObject {
     }
     
     func downloadTemplate(template: CollectedTemplate) {
-        var template = template
-        self.downloadTemplate(template: &template)
-        self.templateItemsViewModel.first(where: {template.templateId == $0.id})?.isDownloaded = true
+        self.isLoading = true
+        entityFetcher?.fetchRelationshipEntities(template: template) { relationships in
+            template.relationships = relationships
+            self.saveTemplate(template: template)
+            self.isLoading = false
+            self.templateItemsViewModel.first(where: {template.templateId == $0.id})?.isDownloaded = true
```

Comment:
Further more Updating the UI elements (isLoading and isDownloaded) should happen on the main thread to avoid potential thread conflicts ? what do you think ?